src/main/java/com/github/dcysteine/neicustomdiagram/api/diagram/DiagramGroup.java

```diff
@@ -288,9 +288,10 @@ public boolean mouseScrolled(GuiRecipe<?> gui, int scroll, int recipe) {
                 diagramState.scroll(direction);
                 return true;
             }
-            if (!ConfigOptions.DISABLE_PAGE_SCROLL.get()) {
```

Comment:
As a separate point, I believe that the logic here is also incorrect. Specifically, this block:

```java
            if (!ConfigOptions.DISABLE_PAGE_SCROLL.get()) {
                return scrollManager.mouseScroll(direction);
            }
```

Here is how the code originally was: \
https://github.com/GTNewHorizons/nei-custom-diagram/blob/d03e63faf8cf78b3a0daa32e2f85039729c9afaf/src/main/java/com/github/dcysteine/neicustomdiagram/api/diagram/DiagramGroup.java#L291-L295

The way that it was supposed to work is, first we'll try to scroll (via calling `scrollManager.mouseScroll()`). If that did anything (i.e. returns `true`), then we will also return `true` and thus consume the event, having successfully scrolled. If it returns `false`, then we'll check the value of the `DISABLE_PAGE_SCROLL` option and return `true` to consume the event if the option is enabled, thus preventing the page scroll handler from receiving the event.

I believe the current behavior is, if the `DISABLE_PAGE_SCROLL` option is enabled, then we will never call `scrollManager.mouseScroll()` at all, and thus never scroll the page. We will also return `false`, and thus not consume the event, allowing the page scroll handler to receive the event even though the option is meant to disable it.

If we want to be succinct, I believe that this will replicate the original behaviour, though in my opinion it is a little harder to read than the original:

```java
    return scrollManager.mouseScroll(direction) || ConfigOptions.DISABLE_PAGE_SCROLL.get();
```

Reply:
Actually, now that I re-read this, I think the problem may be that "page scrolling" refers to two different things here:

1. Scrolling the diagram horizontally or vertically, because it doesn't fit in the window
2. Scrolling through different NEI pages (e.g. different ores for the ore processing diagram)

The `DISABLE_PAGE_SCROLL` option is meant to disable the second one. Sorry, it should maybe be renamed or something.

Reply:
This was intentional. The way you describe it, it's not really a problem with pre-GTNH NEI, however, with the addition of the recipe handler tabs, things change a bit: the thing is, page scrolling is something local to the individual recipe handler, whereas the tab bar is global and should not be affected by handler-specifc settings/behavior (unless you want `DISABLE_PAGE_SCROLL` to act globally, but then that should've been added to NEI itself). In order to ensure that, this setting needs to be bounded to the recipe handler itself, otherwise it will activate with the mouse cursor anywhere in the NEI recipe area including areas that are not part of the handler itself and expect different handling of the scroll wheel.

If this setting is enabled, scrolling through recipe handlers via the tab bar will break for all nei-custom-diagram handler (but work anywhere else).

Reply:
I was typing my response before your second comments This makes sense now after your clarification.